File: solutions/php/lasagna/1/Lasagna.php
<?php

declare(strict_types=1);

final readonly class Lasagna
{
    private const COOK_TIME = 40;
    private const MINUTES_PER_LAYER = 2;
    
    public function expectedCookTime(): int
    {
        return self::COOK_TIME;
    }

    public function remainingCookTime(int $elapsed_minutes): int
    {
        return self::COOK_TIME - $elapsed_minutes;
    }

    public function totalPreparationTime(int $layers_to_prep): int
    {
        return $layers_to_prep * self::MINUTES_PER_LAYER;
    }

    public function totalElapsedTime(int $layers_to_prep, int $elapsed_minutes): int
    {
        return $this->totalPreparationTime($layers_to_prep) + $elapsed_minutes;
    }

    public function alarm(): string
    {
        return 'Ding!';
    }
}

File: solutions/elixir/lasagna/1/lib/lasagna.ex
defmodule Lasagna do
  @minutes_per_layer 2
  @expected_minutes_in_oven 40

  def expected_minutes_in_oven(), do: @expected_minutes_in_oven


  def remaining_minutes_in_oven(current_time_in_oven) do
    expected_minutes_in_oven() - current_time_in_oven
  end

  def preparation_time_in_minutes(layers) do
    layers * @minutes_per_layer
  end

  def total_time_in_minutes(layers, current_time_in_oven) do
    preparation_time_in_minutes(layers) + current_time_in_oven
  end

  def alarm(), do: "Ding!"
end
